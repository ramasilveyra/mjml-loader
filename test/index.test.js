import fs from 'fs';
import path from 'path';
import mjmlLoader from '../src';

const validMJMLPath = path.resolve(__dirname, 'fixtures/valid.mjml');
const validMJML = fs.readFileSync(validMJMLPath, 'utf-8');
const invalidMJMLPath = path.resolve(__dirname, 'fixtures/invalid.mjml');
const invalidMJML = fs.readFileSync(invalidMJMLPath, 'utf-8');

describe('while using mjml-loader', () => {
  it('should convert valid MJML to CommonJS module exporting the HTML result', () => {
    const context = {
      cacheable: jest.fn()
    };

    const result = mjmlLoader.call(context, validMJML);

    expect(result).toMatchSnapshot();
    expect(context.cacheable).toHaveBeenCalled();
  });

  it('should throw with invalid MJML', () => {
    const context = {
      cacheable: jest.fn(),
      resourcePath: invalidMJMLPath
    };

    const result = mjmlLoader.bind(context, invalidMJML);

    expect(result).toThrowErrorMatchingSnapshot();
    expect(context.cacheable).toHaveBeenCalled();
  });
});
