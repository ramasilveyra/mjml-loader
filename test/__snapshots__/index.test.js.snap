// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while using mjml-loader should convert valid MJML to CommonJS module exporting the HTML result 1`] = `"export default \\"<!doctype html>\\\\n<html xmlns=\\\\\\"http://www.w3.org/1999/xhtml\\\\\\" xmlns:v=\\\\\\"urn:schemas-microsoft-com:vml\\\\\\" xmlns:o=\\\\\\"urn:schemas-microsoft-com:office:office\\\\\\">\\\\n<head>\\\\n  <title></title>\\\\n  <!--[if !mso]><!-- -->\\\\n  <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n  <!--<![endif]-->\\\\n<meta http-equiv=\\\\\\"Content-Type\\\\\\" content=\\\\\\"text/html; charset=UTF-8\\\\\\">\\\\n<meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width, initial-scale=1.0\\\\\\">\\\\n<style type=\\\\\\"text/css\\\\\\">\\\\n  #outlook a { padding: 0; }\\\\n  .ReadMsgBody { width: 100%; }\\\\n  .ExternalClass { width: 100%; }\\\\n  .ExternalClass * { line-height:100%; }\\\\n  body { margin: 0; padding: 0; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; }\\\\n  table, td { border-collapse:collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt; }\\\\n  img { border: 0; height: auto; line-height: 100%; outline: none; text-decoration: none; -ms-interpolation-mode: bicubic; }\\\\n  p { display: block; margin: 13px 0; }\\\\n</style>\\\\n<!--[if !mso]><!-->\\\\n<style type=\\\\\\"text/css\\\\\\">\\\\n  @media only screen and (max-width:480px) {\\\\n    @-ms-viewport { width:320px; }\\\\n    @viewport { width:320px; }\\\\n  }\\\\n</style>\\\\n<!--<![endif]-->\\\\n<!--[if mso]>\\\\n<xml>\\\\n  <o:OfficeDocumentSettings>\\\\n    <o:AllowPNG/>\\\\n    <o:PixelsPerInch>96</o:PixelsPerInch>\\\\n  </o:OfficeDocumentSettings>\\\\n</xml>\\\\n<![endif]-->\\\\n<!--[if lte mso 11]>\\\\n<style type=\\\\\\"text/css\\\\\\">\\\\n  .outlook-group-fix {\\\\n    width:100% !important;\\\\n  }\\\\n</style>\\\\n<![endif]-->\\\\n\\\\n<!--[if !mso]><!-->\\\\n    <link href=\\\\\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\\\\\" rel=\\\\\\"stylesheet\\\\\\" type=\\\\\\"text/css\\\\\\">\\\\n    <style type=\\\\\\"text/css\\\\\\">\\\\n\\\\n        @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);\\\\n\\\\n    </style>\\\\n  <!--<![endif]--></head>\\\\n<body>\\\\n  \\\\n  <div style=\\\\\\"cursor:auto;color:#000000;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:22px;text-align:left;\\\\\\">Hello World!</div>\\\\n</body>\\\\n</html>\\";"`;

exports[`while using mjml-loader should throw with invalid MJML 1`] = `
"[mjml-loader] ERROR in /Users/ramasilveyra/dev/personal/mjml-loader/test/fixtures/invalid.mjml:
    - Line 3 (mj-not-valid) â€” Element mj-not-valid doesn't exist or is not registered"
`;
